src/main/java/spireMapOverhaul/patches/interfacePatches/EncounterModifierPatches.java

```diff
@@ -0,0 +1,145 @@
+package spireMapOverhaul.patches.interfacePatches;
+
+import basemod.abstracts.CustomSavable;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePatch2;
+import com.evacipated.cardcrawl.modthespire.lib.SpirePostfixPatch;
+import com.megacrit.cardcrawl.core.Settings;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import com.megacrit.cardcrawl.monsters.MonsterGroup;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.util.ActUtil;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zoneInterfaces.EncounterModifyingZone;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EncounterModifierPatches {
+    // These are deliberately postfix patches so that the original logic still takes place, including advancing through
+    // the list of planned encounters in the act (which means that zone encounters don't delay getting to "hard pool"
+    // normal fights and that you can get the same non-zone encounter if there's a zone encounter in between).
+    @SpirePatch2(clz = AbstractDungeon.class, method = "getMonsterForRoomCreation")
+    public static class NormalEncounterPatch {
+        @SpirePostfixPatch
+        public static MonsterGroup ChangeEncounter(AbstractDungeon __instance, MonsterGroup __result) {
+            MonsterGroup result = __result;
+            AbstractZone zone = Wiz.getCurZone();
+            if (zone instanceof EncounterModifyingZone) {
+                EncounterModifyingZone z = (EncounterModifyingZone)zone;
+                List<EncounterModifyingZone.ZoneEncounter> encounters = z.getNormalEncounters();
+                if (encounters != null) {
+                    encounters = encounters.stream().filter(e -> e.getActNum() == ActUtil.getRealActNum()).collect(Collectors.toList());
+                    if (LastZoneNormalEncounter.lastZoneNormalEncounter != null && LastZoneNormalEncounter.lastZoneNormalEncounter.startsWith(zone.id + ":")) {
```

Comment:
I've built this in under the assumption that all the zone that define their own encounter list will want the standard no-duplicate-encounters behavior. It's something people don't always realize exist, but I think of it as a core part of how encounters work under the hood in Slay the Spire.

If some zone wants to bypass this, it's easy enough for us to add a method to turn it off, but I don't see a reason to encourage it.